share/cmake/modules/Findyaml-cpp.cmake

```diff
@@ -118,51 +124,35 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
         if(PC_yaml-cpp_FOUND)
             set(yaml-cpp_VERSION "${PC_yaml-cpp_VERSION}")
         endif()
+
+        # Create target
+        add_library(yaml-cpp::yaml-cpp UNKNOWN IMPORTED GLOBAL)
+        set_target_properties(yaml-cpp::yaml-cpp PROPERTIES
+            IMPORTED_LOCATION ${yaml-cpp_LIBRARY}
+            INTERFACE_INCLUDE_DIRECTORIES ${yaml-cpp_INCLUDE_DIR}
+        )
+        # Required because Installyaml-cpp.cmake creates `yaml-cpp::yaml-cpp`
+        # as an alias, and aliases get resolved in exported targets.
+        add_library(yaml-cpp ALIAS yaml-cpp::yaml-cpp)
     endif()
 
     # Override REQUIRED if package can be installed
     if(OCIO_INSTALL_EXT_PACKAGES STREQUAL MISSING)
         set(yaml-cpp_FIND_REQUIRED FALSE)
     endif()
 
+    set(YAML_CPP_INCLUDE_DIR "${yaml-cpp_INCLUDE_DIR}")
+    set(YAML_CPP_LIBRARIES yaml-cpp::yaml-cpp)
```

Comment:
Would like to remove the use of `YAML_CPP_LIBRARIES` in the project and just stick to the yaml-cpp::yaml-cpp target now that it is consistently available, could you update that?

Reply:
Doing that would break in case [CMAKE_FIND_PACKAGE_PREFER_CONFIG](https://cmake.org/cmake/help/latest/variable/CMAKE_FIND_PACKAGE_PREFER_CONFIG.html) is enabled and yaml-cpp is pre 0.8. Maybe an acceptable sacrifice?

Reply:
Afraid I'm not following what you mean here, the ALIAS target should be created when version pre 0.8 is found?

Reply:
If `CMAKE_FIND_PACKAGE_PREFER_CONFIG` is enabled this entire module will not be used, and then there also won't be a `yaml-cpp::yaml-cpp` target for v0.7.

Reply:
Thanks @tobim, I was under the assumption that our `ocio_handle_dependency` macro would only use the CONFIG mode if the PREFER_CONFIG argument was specified but it appears the first `find_package` just use the CONFIG mode if `CMAKE_FIND_PACKAGE_PREFER_CONFIG` is specified. If we wanted to force it to use the MODULE first we would have to set `ocio_dep_CONFIG_string` to MODULE by default I assume. @cedrik-fuoco-adsk is this on purpose?

Reply:
I propose to keep `YAML_CPP_LIBRARIES` for now and clean that up at a later time. I can add a TODO comment to that end in this place if you wish?

Reply:
@tobim I agree, let's leave it here for now.